docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,251 @@ Note that:
+
+* The full `path` to the file. That path is treated as path pointing into the remote filesystem of the child-device.
+* An optional configuration `type`. If not provided, the `path` is used as `type`.
+* The fields `user`, `group` and octal `mode` (as introduced in section [Configuration](#configuration) above) are not supported for child-devices.  
+
+The `c8y_configuration_plugin` make use of the HTTP file transfer feature of the `tedge_agent` to consume/provide the configuration-file from/to the child-device's remote filesystem `path` (see [section below](#details-to-aspect-2-filetransfer-fromto-external-device) for more details about HTTP file transfer).
+
+The `c8y_configuration_plugin` adds per child-device the file `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml` implicitely to the child-devices configuration file list. So the list can always be configured from the cloud. The `type` for this self configuration file is `c8y-configuration-plugin`.
+
+### Announcing list of supported configuration files by the child-device itself
+A child-device can use the child-device provisioning API to announce its supported configuration files itself via MQTT to the `c8y_configuration_plugin`. For details about the child-device provisioning API see the [child-device reference documentation](child-devices.md#1-child-device-provisioning).
+
+The provisioning API's `capability specific JSON object` for `c8y_configuration_plugin` contains all configuration files the child-device provides. Each configuration file is defined by a `path` and an optional `type`, that will be both assigned to the fields `path` and `type` of the child-devices TOML file (see section above). 
+
+The MQTT message is as below:
+
+```json
+{
+   "configurations": [
+     {
+       "path": "<filepath 1>",
+       "type": "<config type 1>"
+     },
+     {
+       "path": "<filepath 2>",
+       "type": "<config type 2>"
+     },
+     ...
+   ]
+}
+```
+
+Example:
+```json
+{
+   "configurations": [
+     {
+       "path": "/path/to/file/foo.conf"
+       "type": "foo.conf"
+     },
+     {
+       "path": "/another/path/to/file/bar.conf"     
+     }
+   ]
+}
+```
+
+Whenever the `c8y_configuration_plugin` receivces that MQTT message it stores all contained information to the child-devices individual TOML file in `/etc/tedge/c8y/<childid>/c8y-configuration-plugin.toml`. When the file already exists it will be replaced with the new content.
+
+Each time a child-device's TOML is modified (e.g. due to an incoming MQTT message, an incoming new configuration snapshot from the cloud, or a modification by a local process) the `c8y_configuration_plugin` takes care to define all necessary capabilities to the coresponding cloud's child-device twin. These are:
+  - declaring _supported operations_ for configuration management: `c8y_UploadConfigFile` and `c8y_DownloadConfigFile`
```

Comment:
The supported operations file need to be declared only once, right? The very first time this TOML file is created. Only the `supported configuration types` need to be updated on every update of this TOML file.

Reply:
I changed wording. See https://github.com/cstoidner/thin-edge.io/commit/843cfeea9d24518b4a03597c56c53ccedaa3e5ae